### ISSUE TITLE
question validation error messages are not readable to a screen reader

### ISSUE THREAD

**Issue by bion** [civiform/civiform#1878]

**Comment by goldblatt** [comment]
@clouser2 the a11y bug i mentioned around the client side validation

**Comment by clouser2** [comment]
At this point, there is very little client-side validation (removed as part of #1944). Can someone provide a bit more context as to how to reproduce this error?

I enabled VoiceOver on my Mac, created a form validation error, and the reader announced the error text. As I'm still learning how to use VoiceOver, I'm curious as to what we're expecting to occur differently from what's happening right now.

**Comment by clouser2** [comment]
@bion Are you able to provide more context on how to reproduce the issue described here?

**Comment by clouser2** [comment]
Sounds good. Regarding field association, I believe we can achieve this by adding the `aria-error` and `aria-describedby` attributes to the error / field ([W3C docs](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA21)).

I think this is something we can generally add to FieldWithLabel. The only thing that we'd need to change is that `aria-describedby` requires an actual ID in the DOM. We'd need to add a unique ID to our current error div to support this.

**Comment by jeanniefu** [comment]
Actually while testing stuff for https://github.com/seattle-uat/civiform/issues/1879, I realized we are only reading out the individual field-level errors within a question, but not the broader question-level errors. We should include both in the aria-describedby attribute. See this screenshot for what I mean:
<img width="1582" alt="Screen Shot 2022-05-24 at 5 42 40 PM" src="https://user-images.githubusercontent.com/9094240/170154566-418ff89d-2f18-4231-9794-23a450673e92.png">

Also some types (e.g. radio, checkbox) don't use the underlying FieldWithLabel class, so we'll need some custom work there to tie the error messages to the inputs.

I can take a look at these remaining corner cases since it's related to the other issue (https://github.com/seattle-uat/civiform/issues/1879) I'm looking into!
